src/circular-queue.ts

```diff
@@ -53,8 +64,8 @@ export default class ArmorCircularQueue<T> implements ArmorCollection<T> {
 		} else {
 			result = state;
 
-			if(!this.isValidState(result)){
-				throw new Error('options.state contains errors')
+			if (!this.isValidState(result)) {
+				throw new Error('options.state contains errors');
```

Comment:
Provide specific error messages when throwing so the developer has a clear idea of what went wrong. Given you return null in most places `isValidState` is used (which is good), I recommend replacing the single throw inside this condition with the commented out block of throws below.